profiling/src/profile/api.rs

```diff
@@ -240,32 +226,29 @@ fn function_fetch(pprof: &pprof::Profile, id: u64) -> anyhow::Result<Function> {
     }
 }
 
-fn lines_fetch<'a>(
-    pprof: &'a pprof::Profile,
-    lines: &'a [pprof::Line],
-) -> anyhow::Result<Vec<Line<'a>>> {
-    let mut output = Vec::with_capacity(lines.len());
-    for line in lines {
-        output.push(Line {
-            function: function_fetch(pprof, line.function_id)?,
-            line: line.line,
-        });
-    }
-    Ok(output)
-}
-
 fn location_fetch(pprof: &pprof::Profile, id: u64) -> anyhow::Result<Location> {
     if id == 0 {
         return Ok(Location::default());
     }
 
     match pprof.locations.iter().find(|item| item.id == id) {
```

Comment:
I was a bit confused what this code was doing (and that iterating locations to find an id seems weird since supposedly the ids are sequential and whatnot), but then I realized that this is only used for `impl<'a> TryFrom<&'a pprof::Profile> for Profile<'a>`, which is definitely not something we use outside of tests (...right?)

Would it be worth adding a `#[cfg(test)]` around most of these methods, as a way of documenting "this is not production code"?

Reply:
Prior to this R&D week, I would have said that's fine, but the new profiling replayer added during R&D week uses this function ^_^

But we could a note of some kind.

Reply:
Yeah, I think it'd helpful to mark this code as "not used/should not be used in production"; as that'd make it clear that improvements here are more of a "maybe some day" vs "woah maybe we're missing something".